File: jobs/2026-04-24__19-06-59/b2b_org_settings__7kk2LHB/artifacts/code/update_org.js
const stytch = require('stytch');

const STYTCH_PROJECT_ID = process.env.STYTCH_PROJECT_ID;
const STYTCH_SECRET = process.env.STYTCH_SECRET;
const organization_id = process.argv[2];

if (!STYTCH_PROJECT_ID || !STYTCH_SECRET) {
  console.error("Missing STYTCH_PROJECT_ID or STYTCH_SECRET environment variables.");
  process.exit(1);
}

if (!organization_id) {
  console.error("Missing organization_id as the first command-line argument.");
  process.exit(1);
}

const client = new stytch.B2BClient({
  project_id: STYTCH_PROJECT_ID,
  secret: STYTCH_SECRET,
});

async function updateOrganization() {
  try {
    await client.organizations.update({
      organization_id,
      auth_methods: 'RESTRICTED',
      allowed_auth_methods: ['sso', 'magic_link'],
      email_jit_provisioning: 'RESTRICTED',
    });
    console.log("Organization updated successfully");
  } catch (error) {
    console.error("Error updating organization:", error);
    process.exit(1);
  }
}

updateOrganization();

File: jobs/2026-04-26__00-46-57/b2b_update_organization__GcRxW7L/artifacts/code/update_org.py
import os
import stytch

def main():
    project_id = os.getenv("STYTCH_B2B_PROJECT_ID")
    secret = os.getenv("STYTCH_B2B_SECRET")

    if not project_id or not secret:
        print("Missing STYTCH_B2B_PROJECT_ID or STYTCH_B2B_SECRET environment variables")
        return

    client = stytch.B2BClient(
        project_id=project_id,
        secret=secret,
    )

    try:
        with open("/logs/artifacts/trial_id", "r") as f:
            trial_id = f.read().strip()
    except FileNotFoundError:
        print("trial_id file not found")
        return

    org_name = f"test-org-{trial_id}"
    org_slug = f"test-org-{trial_id}"

    print(f"Creating organization: {org_name}")
    try:
        create_resp = client.organizations.create(
            organization_name=org_name,
            organization_slug=org_slug,
        )
        organization_id = create_resp.organization.organization_id
        print(f"Created organization with ID: {organization_id}")
    except Exception as e:
        print(f"Error creating organization: {e}")
        return

    print(f"Updating organization settings for: {organization_id}")
    try:
        update_resp = client.organizations.update(
            organization_id=organization_id,
            auth_methods="RESTRICTED",
            allowed_auth_methods=["sso", "magic_link"],
        )
        print("Successfully updated organization settings")
        print(f"Auth Methods: {update_resp.organization.auth_methods}")
        print(f"Allowed Auth Methods: {update_resp.organization.allowed_auth_methods}")
    except Exception as e:
        print(f"Error updating organization: {e}")

if __name__ == "__main__":
    main()
